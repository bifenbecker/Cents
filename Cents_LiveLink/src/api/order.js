import httpClient from "./httpClient";

export const fetchStoreSettings = params => {
  return httpClient({
    method: "GET",
    url: "/live-status/settings",
    params,
  });
};

export const authenticateOrder = token => {
  return httpClient({
    method: "GET",
    url: "/live-status/verify-order",
    params: {
      token,
    },
  });
};

export const fetchOrderDetail = token => {
  return httpClient({
    method: "GET",
    url: "/live-status",
    params: {
      token,
    },
  });
};

export const updateTip = (token, tipData) => {
  return httpClient({
    method: "PATCH",
    url: "/live-status/tip",
    params: {
      token,
    },
    data: tipData,
  });
};

export const addPromo = (token, promoData) => {
  return httpClient({
    method: "PATCH",
    url: "/live-status/add-promotion",
    params: {
      token,
    },
    data: promoData,
  });
};

export const removePromo = token => {
  return httpClient({
    method: "PATCH",
    url: "/live-status/remove-promotion",
    params: {
      token,
    },
  });
};

export const addCredits = (token, creditsData) => {
  return httpClient({
    method: "PATCH",
    url: "/live-status/add-credits",
    params: {
      token,
    },
    data: creditsData,
  });
};

export const removeCredits = token => {
  return httpClient({
    method: "PATCH",
    url: "/live-status/remove-credits",
    params: {
      token,
    },
  });
};

export const setOnlineOrderReturnMethod = (token, data) => {
  return httpClient({
    method: "PUT",
    url: "/live-status/return-method",
    params: {
      token,
    },
    data,
  });
};

export const updatePaymentMethodIntent = (token, {paymentToken}) => {
  return httpClient({
    method: "PUT",
    url: "/live-status/payment-method",
    params: {
      token,
    },
    data: {paymentToken},
  });
};

export const cancelOrder = params => {
  return httpClient({
    method: "PATCH",
    url: "/live-status/order/cancel",
    params,
  });
};

export const getRouteDelivery = (token, {orderDelivery}) => {
  return httpClient({
    method: "GET",
    url: `/live-status/order-delivery/${orderDelivery}/route-details`,
    params: {token},
  });
};

export const fetchOrderDeliveries = token => {
  return httpClient({
    method: "GET",
    url: `/live-status/order-deliveries`,
    params: {token},
  });
};

export const manageOrder = (token, data) => {
  return httpClient({
    method: "PUT",
    url: `/live-status/live-link/manage`,
    params: {token},
    data,
  });
};
