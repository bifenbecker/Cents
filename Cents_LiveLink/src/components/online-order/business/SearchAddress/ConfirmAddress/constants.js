export const PLACE_HOLDER = {
  ADDRESS_DETAILS: "Address details",
  PICKUP_INSTRUCTIONS:
    "e.g. call upon arrival, entrance on right side of building, don’t ring bell, etc.",
};

export const LABEL_TEXT = {
  ADDRESS_DETAILS: "Apt / Suite / Unit",
  PICKUP_INSTRUCTIONS: "Pickup Instructions",
};

export const SUBTITLE_TEXT = {
  ADDRESS_DETAILS: "Confirm your address for pickup",
  PICKUP_INSTRUCTIONS: "Pickup Instructions",
};

export const CHECKBOX_LABEL = "Leave it at my door if I’m not home";

export const GET_BTN_TEXT = (flag) => {
  return flag ? "Save & Continue" : "Save";
};

export const NOTES_TEXT = "We will send you updates about your pickup via text.";
