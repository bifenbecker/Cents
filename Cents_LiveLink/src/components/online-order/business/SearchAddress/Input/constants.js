export const SEARCH_FOR_A_NEW_ADDRESS = "Search for a new address";
export const SEARCH_FOR_AN_ADDRESS = "Search for an address";
export const SEARCH_ADDRESS = "Search Address";
