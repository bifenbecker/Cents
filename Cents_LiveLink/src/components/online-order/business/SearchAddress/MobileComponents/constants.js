export const STATES = {
  CONFIRM: "CONFIRM",
  CANT_PICKUP: "CANT_PICKUP",
  AUTO_COMPLETE: "AUTO_COMPLETE",
  SAVED_ADDRESSES: "SAVED_ADDRESSES",
};
