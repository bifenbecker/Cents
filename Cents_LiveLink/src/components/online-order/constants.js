export const orderPickUpSteps = {
  details: "order-details",
  finishingUp: "finishing-up",
};

export const deliveryProviders = {
  uber: "UBER",
  ownDriver: "OWN_DRIVER",
  doorDash: "DOORDASH",
};

export const orderFields = [
  "servicePriceId",
  "serviceModifierIds",
  "customerNotes",
  "isHangDrySelected",
  "hangDryInstructions",
  "orderNotes",
  "customerAddressId",
  "paymentToken",
  "promoCode",
  "bagCount",
  "returnMethod",
  "hasDryCleaning",
];

export const mandatoryOrderFields = [
  "paymentToken",
  "returnMethod",
  "servicePriceId",
  "customerAddressId",
];

export const bufferRequiredForOrder = {
  [deliveryProviders.uber]: 10,
  [deliveryProviders.doorDash]: 10,
  [deliveryProviders.ownDriver]: 20,
};

export const orderDeliveryFields = ["pickup", "delivery"];

export const orderDeliverySubFields = [
  "type",
  "timingsId",
  "deliveryProvider",
  "deliveryWindow",
  "thirdPartyDeliveryId",
  "totalDeliveryCost",
  "courierTip",
  "subsidyInCents",
  "thirdPartyDeliveryCostInCents",
];

export const onDemandDeliveryTypes = {
  pickup: "PICKUP",
  delivery: "DELIVERY",
  pickupAndDelivery: "PICKUP_AND_DELIVERY",
};

export const SERVICE_CATEGORY_TYPES = {
  DRY_CLEANING: "DRY_CLEANING",
  LAUNDRY: "LAUNDRY",
  ALTERATIONS: "ALTERATIONS",
};

export const cloneOrderState = {
  REVIEW: "REVIEW_CLONE_ORDER",
  EDIT: "EDIT_CLONE_ORDER",
  INITIAL: "NO_CLONE_ORDER",
};
