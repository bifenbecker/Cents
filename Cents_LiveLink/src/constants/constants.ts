export const OTP_CODE_LENGTH = 6;
export const PHONE_LENGTH = 10;

export const PHONE_NUMBER_FORMAT_REGEXP = /^(\d{3})(\d{3})(\d{4})$/;
export const NOT_DIGITS_REGEXP = /\D/g;
