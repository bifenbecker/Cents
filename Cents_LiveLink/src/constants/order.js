export const TRANSITION_TIME = 300;

export const DELIVERY_TRACKING_ORDER_STATUSES = {
  submitted: "SUBMITTED",
  intentCreated: "INTENT_CREATED",
  scheduled: "SCHEDULED",
  enRouteToPickup: "EN_ROUTE_TO_PICKUP",
  enRouteToDropoff: "EN_ROUTE_TO_DROPOFF",
  completed: "COMPLETED",
  canceled: "CANCELLED",
};

export const VIEWS = {
  RECOMMENDED_PICKUP: "RECOMMENDED_PICKUP",
  ALL_WINDOWS_PICKUP: "ALL_WINDOWS_PICKUP",
  RETURN_QUESTION: "RETURN_QUESTION",
  SUBSCRIPTION_OFFER: "SUBSCRIPTION_OFFER",
  RECOMMENDED_RETURN: "RECOMMENDED_RETURN",
  ALL_WINDOWS_RETURN: "ALL_WINDOWS_RETURN",
  AVAILABLE_PICKUP_DATES: "AVAILABLE_PICKUP_DATES",
  AVAILABLE_RETURN_DATES: "AVAILABLE_RETURN_DATES",
};

export const DELIVERY_TYPE_KEYS = {
  OWN: "ownDelivery",
  ON_DEMAND: "onDemandDelivery",
};

export const UBER_DELIVERY_STATUS_INDICES = {
  [DELIVERY_TRACKING_ORDER_STATUSES.scheduled]: 0,
  [DELIVERY_TRACKING_ORDER_STATUSES.enRouteToPickup]: 1,
  [DELIVERY_TRACKING_ORDER_STATUSES.enRouteToDropoff]: 2,
  [DELIVERY_TRACKING_ORDER_STATUSES.completed]: 3,
};

export const OWN_DRIVER_DELIVERY_STATUS_INDICES = {
  [DELIVERY_TRACKING_ORDER_STATUSES.scheduled]: 0,
  [DELIVERY_TRACKING_ORDER_STATUSES.enRouteToDropoff]: 1,
  [DELIVERY_TRACKING_ORDER_STATUSES.completed]: 2,
};

export const DELIVERY_PROVIDERS = {
  uber: "UBER",
  ownDriver: "OWN_DRIVER",
  doorDash: "DOORDASH",
};

export const ORDER_DELIVERY_TYPES = {
  pickup: "PICKUP",
  return: "RETURN",
};

export const RETURN_METHODS = {
  delivery: "DELIVERY",
  inStorePickup: "IN_STORE_PICKUP",
};

export const ORDER_DELIVERY_UPDATABLE_STATUSES = [
  DELIVERY_TRACKING_ORDER_STATUSES.intentCreated,
  DELIVERY_TRACKING_ORDER_STATUSES.scheduled,
];

export const ORDER_TYPES = {
  residential: "RESIDENTIAL",
  service: "SERVICE",
  online: "ONLINE",
};

export const COMPLETED_OR_CANCELED_ORDER_STATUSES = ["COMPLETED", "CANCELLED"];

export const INITIAL_PICKUP_AND_DELIVERY_STATE = {
  type: null,
  storeId: null,
  pickupAt: null,
  timingsId: null,
  selectedDate: null,
  uberEstimate: null,
  deliveryWindow: [],
  totalDeliveryCost: 0,
  doorDashEstimate: null,
  deliveryProvider: null,
  thirdPartyDeliveryId: null,
};

export const orderChoices = {
  currentOrder: "currentOrder",
  currentAndFutureOrders: "currentAndFutureOrders",
};

export const orderChoicesDisplay = {
  currentOrder: "This order only",
  currentAndFutureOrders: "This and future recurring orders",
};

export const TEXT_ME_WHEN_READY = "Text me when it's ready";
export const IN_STORE_PICKUP = "In-Store Pickup";
export const WE_WILL_TEXT_YOU_WHEN_READY = "We’ll text you when it’s ready.";

export const deliveryProviders = {
  uber: "UBER",
  ownDriver: "OWN_DRIVER",
  doorDash: "DOORDASH",
};

export const SCHEDULE_TABS = {
  economy: "Economy",
  flexPickup: "Flex Pickup",
};

export const NAMED_DAYS = {
  today: "Today",
  tomorrow: "Tomorrow",
};

export const CENTS_IN_DOLLAR = 100;

export const SCHEDULE = {
  pickupTitle: "Schedule Pickup",
  returnDelivery: "Return delivery",
  pickupSubTitle: "Schedule your laundry to be picked up:",
  availableTimesTitle: "More Available Times",
  setPickupTime: "Set pickup time",
  setReturnTime: "Schedule pickup & delivery",
  recommended: "Recommended",
  moreTimes: "More Available Times",
  seeAllTimes: "See all pickup times",
  recurringOrder: "Recurring Order",
  pickup: "Pickup:",
  delivery: "Delivery:",
  pickupScheduled: "Pickup Scheduled:",
  selectPickupDate: "Select pickup date",
  selectReturnDate: "Select return date",
};

export const ORDER_MESSAGES = {
  scheduleNow: "I'll schedule my return delivery now",
  scheduleDeliveryNow: "Schedule delivery now",
  scheduleDeliveryLater: "Schedule delivery later",
  getTurnAroundInHours: (turnAroundInHours) => `est. ${turnAroundInHours} hrs`,
  getSubscriptionDiscount: (discount) => `SAVE ${discount}% ON EVERY ORDER`,
  recurringCall: "Save loads of time! Schedule your recurring pickup today.",
  pickupReminder:
    "You’ll receive a reminder to get your laundry ready the night before we are scheduled to pick it up. ",
  noTimes: "No Available times",
  trySelectFlex: "Try selecting Flex or a different day",
  trySelectEconomy: "Try selecting Economy or a different day",
};

export const COLORS = {
  background: "#F5F5F5",
  containerBackground: "#FFFFFF",
};

export const TYPOGRAPHY = {
  default:
    "Inter, Roboto Bold, Roboto Regular, -apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Oxygen, Ubuntu, Cantarell, Fira Sans, Droid Sans, Helvetica Neue, sans-serif",
};

export const SCHEDULE_BUTTONS = {
  edit: "Edit",
  scheduleDeliveryNow: "Schedule delivery now",
  editScheduling: "Edit scheduling",
  notNow: "Not now",
  apply: "Apply",
  setPickup: "Set pickup time",
  confirm: "Confirm",
  cancel: "Cancel",
};
