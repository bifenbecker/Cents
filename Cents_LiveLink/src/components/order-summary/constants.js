import {RETURN_METHODS} from "../../constants/order";
import {
  CALL_ICON,
  DELIVERY_ORDER_COMPLETED,
  ON_DEMAND_DELIVERY_IN_PROGRESS,
  ON_DEMAND_PICKUP_COMPLETED,
  ON_DEMAND_PICKUP_INITIATED,
  ON_DEMAND_PICKUP_SCHEDULED,
  ORDER_CANCELED,
  ORDER_COMPLETED,
  PROCESSING,
  READY_FOR_CUSTOMER_PICKUP,
  SMS_ICON,
  STD_DELIVERY_IN_PROGRESS,
  STD_PICKUP_COMPLETED,
  STD_PICKUP_INITIATED,
  STD_PICKUP_SCHEDULED,
} from "../../assets/images";

export {RETURN_METHODS, ORDER_TYPES} from "../../constants/order";

export const RESIDENTIAL_ORDER_STATUSES = {
  submitted: "Submitted",
  processing: "Received & Processing",
  delivering: "Delivering to you",
  completed: "Complete",
};

export const ONLINE_ORDER_STATUSES = {
  submitted: "Submitted",
  processing: "Received & Processing",
  readyForPickup: "Laundry Ready",
  delivering: "Delivering to you",
  completed: "Complete",
};

export const SERVICE_ORDER_STATUSES = {
  submitted: "Submitted",
  processing: "Processing",
  readyForPickup: "Laundry Ready",
  completed: "Complete",
};

export const DELIVERY_ORDER_STATUSES = {
  submitted: "Submitted",
  processing: "Processing",
  delivering: "Delivering to you",
  completed: "Complete",
};

export const TIMELINE_CANCELED_STATUS = "Canceled";

export const DELIVERY_TRACKING_ORDER_STATUSES = {
  submitted: "SUBMITTED",
  enRouteToPickup: "EN_ROUTE_TO_PICKUP",
  enRouteToDropoff: "EN_ROUTE_TO_DROPFF",
  completed: "COMPLETED",
};

export const RETURN_METHODS_DISPLAY = {
  [RETURN_METHODS.inStorePickup]: "NO THANKS",
  [RETURN_METHODS.delivery]: "DELIVER TO ME",
};

export const serviceSelections = {
  laundry: "Laundry",
  dryCleaning: "Dry Cleaning",
};

export const orderItemCategories = {
  deliveryOrPickup: "DELIVERY",
  fixedPriceItem: "FIXED_PRICE",
  perPoundItem: "PER_POUND",
};

export const ORDER_STATUSES = {
  COMPLETED: "COMPLETED",
  CANCELLED: "CANCELLED",
};

export const IMAGE_IDS = {
  DELIVERY_ORDER_COMPLETED: DELIVERY_ORDER_COMPLETED,
  ON_DEMAND_DELIVERY_IN_PROGRESS: ON_DEMAND_DELIVERY_IN_PROGRESS,
  ON_DEMAND_PICKUP_COMPLETED: ON_DEMAND_PICKUP_COMPLETED,
  ON_DEMAND_PICKUP_INITIATED: ON_DEMAND_PICKUP_INITIATED,
  ON_DEMAND_PICKUP_SCHEDULED: ON_DEMAND_PICKUP_SCHEDULED,
  ORDER_CANCELED: ORDER_CANCELED,
  ORDER_COMPLETED: ORDER_COMPLETED,
  PROCESSING: PROCESSING,
  READY_FOR_CUSTOMER_PICKUP: READY_FOR_CUSTOMER_PICKUP,
  STD_DELIVERY_IN_PROGRESS: STD_DELIVERY_IN_PROGRESS,
  STD_PICKUP_COMPLETED: STD_PICKUP_COMPLETED,
  STD_PICKUP_SCHEDULED: STD_PICKUP_SCHEDULED,
  STD_PICKUP_INITIATED: STD_PICKUP_INITIATED,
  CALL_ICON,
  SMS_ICON,
};

export const PAYMENT_STATUSES = {
  balanceDue: "BALANCE_DUE",
  paid: "PAID",
  pending: "PENDING",
};
