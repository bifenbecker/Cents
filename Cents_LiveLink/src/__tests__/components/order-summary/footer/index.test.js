import React from "react";
import {fireEvent, render} from "@testing-library/react";

import Footer from "../../../../components/order-summary/footer";
import {
  ORDER_STATUSES,
  PAYMENT_STATUSES,
} from "../../../../components/order-summary/constants";
import {DELIVERY_TRACKING_ORDER_STATUSES, ORDER_TYPES} from "../../../../constants/order";

const renderFooter = props => {
  const {getByText, container} = render(<Footer {...props} />);

  return {getByText, container};
};

describe("Footer Component", () => {
  describe("Drawer is not open", () => {
    it("should show nothing", () => {
      const {container} = renderFooter({drawerOpen: false});
      expect(container.firstChild).toBeEmptyDOMElement();
    });
  });

  describe("Drawer is open", () => {
    describe("Order is CANCELLED", () => {
      const orderDetails = {status: ORDER_STATUSES.CANCELLED, refundableAmount: 10};
      it("should match snapshot", () => {
        expect(
          render(
            <Footer
              {...{
                drawerOpen: true,
                orderDetails,
              }}
            />
          )
        ).toMatchSnapshot();
      });

      it("should show how much amount is credited back", () => {
        const {getByText} = renderFooter({drawerOpen: true, orderDetails});

        expect(getByText("Account Credited")).toBeInTheDocument();
        expect(getByText("$10.00")).toBeInTheDocument();
      });
    });

    describe("Order is of type ONLINE", () => {
      const orderDetails = {
        orderType: ORDER_TYPES.online,
        pickup: {status: DELIVERY_TRACKING_ORDER_STATUSES.completed},
        netOrderTotal: 5,
        isIntakeComplete: true,
        paymentStatus: PAYMENT_STATUSES.paid,
      };
      it("should match snapshot", () => {
        expect(
          render(
            <Footer
              {...{
                drawerOpen: true,
                orderDetails,
              }}
            />
          )
        ).toMatchSnapshot();
      });

      it("should show OnlineOrder footer details", () => {
        const {getByText} = renderFooter({
          drawerOpen: true,
          orderDetails,
        });

        expect(getByText("Total Paid")).toBeInTheDocument();
      });
    });

    describe("Order has balanceDue", () => {
      const orderDetails = {
        status: ORDER_STATUSES.COMPLETED,
        balanceDue: 6.78,
        netOrderTotal: 12,
      };

      describe("If there is a selected paymentMethod", () => {
        const paymentMethod = {
          brand: "visa",
          centsCustomerId: 1,
          id: 1,
          last4: "4242",
          paymentMethodToken: "pm_1IqHlhGuj5YLpJjFuq3ft1tty",
          provider: "stripe",
          type: "credit",
        };
        it("should match snapshot", () => {
          expect(
            render(
              <Footer
                {...{
                  drawerOpen: true,
                  orderDetails,
                  paymentMethod,
                }}
              />
            )
          ).toMatchSnapshot();
        });

        it("should have PAY button", () => {
          const {getByText} = renderFooter({
            drawerOpen: true,
            orderDetails,
            paymentMethod,
          });

          expect(getByText("PAY $6.78")).toBeInTheDocument();
        });

        it("should call onPayForOrder on PAY button click", () => {
          const onPayForOrder = jest.fn();
          const {getByText} = renderFooter({
            drawerOpen: true,
            orderDetails,
            paymentMethod,
            onPayForOrder,
          });

          const button = getByText("PAY $6.78");
          fireEvent.click(button);

          expect(onPayForOrder).toBeCalled();
        });
      });

      describe("If there is no selected paymentMethod", () => {
        it("should match snapshot", () => {
          expect(
            render(
              <Footer
                {...{
                  drawerOpen: true,
                  orderDetails,
                }}
              />
            )
          ).toMatchSnapshot();
        });

        it("should have PAY button", () => {
          const {getByText} = renderFooter({
            drawerOpen: true,
            orderDetails,
          });

          expect(getByText("Enter card & pay $6.78")).toBeInTheDocument();
        });

        it("should call onAddPaymentMethod on PAY button click", () => {
          const onAddPaymentMethod = jest.fn();
          const {getByText} = renderFooter({
            drawerOpen: true,
            orderDetails,
            onAddPaymentMethod,
          });

          const button = getByText("Enter card & pay $6.78");
          fireEvent.click(button);

          expect(onAddPaymentMethod).toBeCalled();
        });
      });
    });

    describe("Order does not have balanceDue", () => {
      const orderDetails = {
        status: ORDER_STATUSES.COMPLETED,
        balanceDue: 0,
        netOrderTotal: 12,
      };

      it("should match snapshot", () => {
        expect(
          <Footer
            {...{
              drawerOpen: true,
              orderDetails,
            }}
          />
        ).toMatchSnapshot();
      });

      it("should show how much amount is paid", () => {
        const {getByText} = renderFooter({drawerOpen: true, orderDetails});

        expect(getByText("Total Paid")).toBeInTheDocument();
        expect(getByText("$12.00")).toBeInTheDocument();
      });
    });
  });
});
