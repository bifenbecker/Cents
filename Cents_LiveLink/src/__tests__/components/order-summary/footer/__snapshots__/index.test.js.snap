// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component Drawer is open Order does not have balanceDue should match snapshot 1`] = `
<Footer
  drawerOpen={true}
  orderDetails={
    Object {
      "balanceDue": 0,
      "netOrderTotal": 12,
      "status": "COMPLETED",
    }
  }
/>
`;

exports[`Footer Component Drawer is open Order has balanceDue If there is a selected paymentMethod should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bczRLJ sc-gsnTZi qSOsC hzPtvw"
        width="100%"
      >
        <div
          class="sc-bczRLJ sc-gsnTZi mLOuI hzPtvw"
          height="100%"
          width="100%,100%,100%,50%"
        >
          <button
            class="sc-bczRLJ fjaIru"
            font-size="1.25rem"
            height="56px"
            width="100%"
          >
            PAY
             
            $6.78
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bczRLJ sc-gsnTZi qSOsC hzPtvw"
      width="100%"
    >
      <div
        class="sc-bczRLJ sc-gsnTZi mLOuI hzPtvw"
        height="100%"
        width="100%,100%,100%,50%"
      >
        <button
          class="sc-bczRLJ fjaIru"
          font-size="1.25rem"
          height="56px"
          width="100%"
        >
          PAY
           
          $6.78
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Footer Component Drawer is open Order has balanceDue If there is no selected paymentMethod should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bczRLJ sc-gsnTZi qSOsC hzPtvw"
        width="100%"
      >
        <div
          class="sc-bczRLJ sc-gsnTZi mLOuI hzPtvw"
          height="100%"
          width="100%,100%,100%,50%"
        >
          <button
            class="sc-bczRLJ fjaIru"
            font-size="1.25rem"
            height="56px"
            width="100%"
          >
            Enter card & pay
             
            $6.78
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bczRLJ sc-gsnTZi qSOsC hzPtvw"
      width="100%"
    >
      <div
        class="sc-bczRLJ sc-gsnTZi mLOuI hzPtvw"
        height="100%"
        width="100%,100%,100%,50%"
      >
        <button
          class="sc-bczRLJ fjaIru"
          font-size="1.25rem"
          height="56px"
          width="100%"
        >
          Enter card & pay
           
          $6.78
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Footer Component Drawer is open Order is CANCELLED should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bczRLJ sc-gsnTZi qSOsC hzPtvw"
        width="100%"
      >
        <div
          class="sc-bczRLJ sc-gsnTZi dfqnXC hzPtvw"
          color="BLACK"
          font-size="4"
          height="100%"
          width="100%"
        >
          <div
            class="sc-bczRLJ jITePI"
          >
            Account Credited
          </div>
          <div
            class="sc-bczRLJ jITePI"
          >
            $10.00
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bczRLJ sc-gsnTZi qSOsC hzPtvw"
      width="100%"
    >
      <div
        class="sc-bczRLJ sc-gsnTZi dfqnXC hzPtvw"
        color="BLACK"
        font-size="4"
        height="100%"
        width="100%"
      >
        <div
          class="sc-bczRLJ jITePI"
        >
          Account Credited
        </div>
        <div
          class="sc-bczRLJ jITePI"
        >
          $10.00
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Footer Component Drawer is open Order is of type ONLINE should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bczRLJ sc-gsnTZi qSOsC hzPtvw"
        width="100%"
      >
        <div
          class="sc-bczRLJ sc-gsnTZi dfqnXC hzPtvw"
          color="BLACK"
          font-size="4"
          height="100%"
          width="100%"
        >
          <div
            class="sc-bczRLJ jITePI"
          >
            Total Paid
          </div>
          <div
            class="sc-bczRLJ jITePI"
          >
            $5.00
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bczRLJ sc-gsnTZi qSOsC hzPtvw"
      width="100%"
    >
      <div
        class="sc-bczRLJ sc-gsnTZi dfqnXC hzPtvw"
        color="BLACK"
        font-size="4"
        height="100%"
        width="100%"
      >
        <div
          class="sc-bczRLJ jITePI"
        >
          Total Paid
        </div>
        <div
          class="sc-bczRLJ jITePI"
        >
          $5.00
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
